front/pages/w/[wId]/builder/data-sources/index.tsx

```diff
@@ -566,90 +521,105 @@ export default function DataSourcesView({
+            title="Static Data Sources"
+            description="Lets you expose static data to Dust."
+            action={
+              !readOnly
+                ? {
+                    label: "Add a new Data Source",
+                    variant: "secondary",
+                    icon: PlusIcon,
+                    onClick: () => {
+                      // Enforce plan limits: DataSources count.
+                      if (
+                        owner.plan.limits.dataSources.count != -1 &&
+                        dataSources.length >=
+                          owner.plan.limits.dataSources.count
+                      ) {
+                        window.alert(
+                          "You are limited to 1 DataSource on our free plan. Contact team@dust.tt if you want to increase this limit."
+                        );
+                        return;
+                      } else {
+                        void router.push(
+                          `/w/${owner.sId}/builder/data-sources/new`
+                        );
+                      }
+                    },
+                  }
+                : undefined
+            }
+          />
 
-        <div className="my-4">
-          <ul role="list" className="mt-4 divide-y divide-structure-200">
+          <ContextItem.List className="mt-2">
             {dataSources.map((ds) => (
-              <li key={ds.name} className="px-2">
-                <div className="py-4">
-                  <div className="flex items-start">
-                    <div className="mr-2 flex h-5 w-5 flex-initial sm:mr-4">
-                      <CloudArrowDownIcon className="h-5 w-5 text-element-600" />
-                    </div>
-                    <div className="fexl flex-col">
-                      <Link
-                        href={`/w/${owner.sId}/builder/data-sources/${ds.name}`}
-                        className="flex"
-                      >
-                        <p className="truncate text-sm font-bold text-element-900">
-                          {ds.name}
-                        </p>
-                      </Link>
-                      <div className="mt-2 text-sm text-element-700">
-                        {ds.description}
-                      </div>
-                    </div>
-
-                    <div className="flex flex-1"></div>
-
-                    <div>
-                      <Button
-                        variant="secondary"
-                        icon={Cog6ToothIcon}
-                        onClick={() => {
-                          void router.push(
-                            `/w/${owner.sId}/builder/data-sources/${ds.name}`
-                          );
-                        }}
-                        label="Manage"
-                      />
-                    </div>
+              <ContextItem
+                key={ds.name}
+                title={ds.name}
+                visual={<ContextItem.Visual visual={CloudArrowDownIcon} />}
```

Comment:
```suggestion
                visual={
                  <ContextItem.Visual
                    visual={({ className }) =>
                      FolderOpenIcon({
                        className: className + " text-element-600",
                      })
                    }
                  />
```
To match the icon from figma. Somewhat heavy, but otherwise requires to change ContextVisual in sparkle

Reply:
TBD since at the time this is the only place I've seen this need